### ISSUE TITLE
How to differentiate through the MFC pre-processing step?

### ISSUE THREAD

**Issue by pengcheng-tech** [carlini/audio_adversarial_examples#39]
Hi,

Could you please give me a hint where in the code you calculate the gradient descent through the MFC pre-processing transformation?

I am new in this area and I am a bit lost when trying to find the relevant code part.

Thanks

**Comment by carlini** [comment]
The gradient is computed here

https://github.com/carlini/audio_adversarial_examples/blob/bc1f3752e531afdd8e1bb846f4aed2a44ad98e94/attack.py#L133

although the loss function is defined earlier

https://github.com/carlini/audio_adversarial_examples/blob/bc1f3752e531afdd8e1bb846f4aed2a44ad98e94/attack.py#L107-L110

where the logits are actually computed again earlier

https://github.com/carlini/audio_adversarial_examples/blob/bc1f3752e531afdd8e1bb846f4aed2a44ad98e94/attack.py#L97
